.github/workflows/client.yml

```diff
@@ -0,0 +1,81 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows
+on:
+  push:
+    branches:
+      # TODO: Run on master after we're fully migrated from Circle CI
+      # - master
+      - "rfc-18/**"
+  pull_request:
+      branches:
+      # TODO: Run on all branches after we're fully migrated from Circle CI
+      - "rfc-18/**"
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      # TODO: This step was left here intentionally so we can track disk space
+      # usage for a while. We were trying to fight problems with out of disk space
+      # that happened due to the size of data restored from cache. The cache size
+      # was growing linearly with subsequent workflow runs. We want to observe
+      # available disk space for `/`. Fresh execution starts with 21 GB, we expect
+      # to have no less than 15 GB after the cache is restored.
+      - run: sudo df -h
+
+      - name: Build Docker Build Image
+        uses: docker/build-push-action@v2
+        with:
+          target: gobuild
+          tags: go-build-env
+          build-args: |
+            REVISION=${{ github.sha }}
+          # VERSION= ? TODO: Configure version, sample: 1.7.6
+          load: true # load image to local registry to use it in next steps
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+
+      - name: Create test results directory
+        run: |
+          mkdir test-results
+      - name: Run Go tests
+        run: |
+          docker run \
+            --volume $GITHUB_WORKSPACE/test-results:/mnt/test-results \
+            --workdir /go/src/github.com/keep-network/keep-ecdsa \
+            go-build-env \
+            gotestsum --junitfile /mnt/test-results/unit-tests.xml
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action@v1.7
```

Comment:
There's `v1.9` tag available. We could define simply `@v1` and I believe it should pull the latest `v1.x`.